server/src/main/java/io/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -158,29 +179,48 @@ public void start()
     return identifiers;
   }
 
-  private VersionedIntervalTimeline<String, DataSegment> getTimelineForIntervalWithHandle(
+  private VersionedIntervalTimeline<String, DataSegment> getTimelineForIntervalsWithHandle(
       final Handle handle,
       final String dataSource,
-      final Interval interval
+      final List<Interval> intervals
   ) throws IOException
   {
-    final VersionedIntervalTimeline<String, DataSegment> timeline = new VersionedIntervalTimeline<>(
-        Ordering.natural()
-    );
+
+    final StringBuilder sb = new StringBuilder();
+    sb.append("SELECT payload FROM %s WHERE used = true AND dataSource = :dataSource AND (");
+    for (int i = 0; i < intervals.size(); i++) {
+      Interval interval = intervals.get(i);
+      sb.append(
+          "(start <= '"
+          + interval.getEnd().toString()
```

Comment:
Are you 100% sure there is no possibility for any questionable characters to be in here, no matter what the locale or any other settings? If so that is cool (a comment would be great). Could also put stuff here like `:start1` and `:end1` and bind those programmatically.